src/npcmove.cpp

```diff
@@ -3746,23 +3762,35 @@ static float rate_food( const item &it, int want_nutr, int want_quench )
     }
 
     double relative_rot = it.get_relative_rot();
+
+    // Don't eat rotten food.
     if( relative_rot >= 1.0f ) {
         // TODO: Allow sapro mutants to eat it anyway and make them prefer it
         return 0.0f;
     }
 
+    // For non-rotten food, we have a starting weight in the range 1-10
+    // The closer it is to expiring, the more we should aim to eat it.
     float weight = std::max( 1.0, 10.0 * relative_rot );
+
+    // TODO: I feel like we should exclude *really* un-fun foods (flour, hot sauce, etc)
+    //       rather than discount them. Eating cooked liver is fine, eating raw flour... :/
```

Comment:
If they're starving, I agree.

But if the NPC gets peckish on a road trip, having them choke down some raw flour because they can't wait another half-hour until we stop at a FoodPlace™ doesn't feel right.

Separately to this PR, I'd love to make NPC need satisfaction more nuanced.

Reply:
How about a threshold to discard decreasing as the NPC goes from CONSUME to COMPLAIN?

Reply:
Oh, I've got unsubmitted changes whereby NPCs won't eat things which they wouldn't eat if they were placed in a basecamp larder (ie, fun < 6), but I haven't verified they actually work as intended yet.

You can still *manually* ask NPCs to eat flour, but that change (which will be in a separate PR) will stop them from chugging hotsauce during supermarket runs. :)